### PR TITLE
Make Leap Motion a plugin

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1045/HighFidelity-Beta-PR10797-8682cee17ae68a36158cef40e38b0ec28561d41c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1045/HighFidelity-Beta-PR10797-8682cee17ae68a36158cef40e38b0ec28561d41c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1049/HighFidelity-Beta-PR10797-9d0407cbdfa892ccf852bfb656c869b1d1657606.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1049/HighFidelity-Beta-PR10797-9d0407cbdfa892ccf852bfb656c869b1d1657606.exe

**Comment by ctrlaltdavid** [comment]
**Testing**

Leap Motion 2.3.1 runtime installed (i.e., not "Orion" beta).

Run Sandbox and Interface from PR build.

Use an avatar with individual features (e.g., "Being of Light").

Settings > General > Leap Motion > Enabled
- Enables / disables Leap Motion control of hands.
- Setting is remembered between Interface runs.

Settings > General > Leap Motion > Sensor location
- Sets up Interface to use Leap Motion on desktop / HMD.
- Setting is remembered between Interface runs.

Settings > General > Leap Motion > Desktop height for horizontal forearms (default 0.20m)
- Sets the height of your hands above the Leap Motion that corresponds to your avatar's forearms being horizontal.
- Setting is remembered between Interface runs.

Leap Motion controls position and orientation of hands.
Leap Motion controls fingers and thumbs.

When tracking of a hand is lost, your avatar's arm, hand, and fingers return to their relaxed, non-controlled positions.
When returned to relaxed, non-controlled positions, hand and fingers resume with voice-activated gesturing (i.e., slight movements as you speak).

Check another person (running PR Interface) can see your avatars' hands and fingers moving.

Test changing between desktop and HMD operation while Interface is running.

Test connecting Leap Motion after Interface has started.

Test enabling Leap Motion support in Settings and connecting Leap Motion after Interface has started.


**Notes**

There is no "leapHands.js" script any more; you don't need to run a script in order to have Leap Motion control your avatar's hands.

Leap Motion hand and finger tracking isn't perfect. If you want to compare what Interface is showing and what the Leap Motion runtime is detecting, in your Windows system tray, right-click the Leap Motion icon and select "Visualizer...". You can run Interface and the Visualizer side-by-side, though (depending on your Leap Motion configuration) you'll probably need Interface to have focus.
By default you'll need to have Interface 

Should run on Mac OSX as well as Windows (though probably only need to test on Windows as OSX support may be removed in the future if Leap Motion plugin is updated to use Orion SDK).

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1050/HighFidelity-Beta-PR10797-9f48a139cc695ccb3bac784e38fec586be0abda8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1050/HighFidelity-Beta-PR10797-9f48a139cc695ccb3bac784e38fec586be0abda8.exe

File: interface/resources/qml/hifi/tablet/TabletGeneralPreferences.qml [hunk 1]
```diff
@@ -32,6 +32,6 @@ StackView {
     TabletPreferencesDialog {
         id: root
         objectName: "TabletGeneralPreferences"
-        showCategories: ["UI", "Snapshots", "Scripts", "Privacy", "Octree", "HMD", "Sixense Controllers", "Perception Neuron", "Kinect", "Vive Pucks Configuration"]
+        showCategories: ["UI", "Snapshots", "Scripts", "Privacy", "Octree", "HMD", "Sixense Controllers", "Perception Neuron", "Kinect", "Vive Pucks Configuration", "Leap Motion"]
```
Comment (jherico) [inline interface/resources/qml/hifi/tablet/TabletGeneralPreferences.qml]: not a blocker on this PR, but this model isn't sustainable.  We need to find some way for plugins to inject preferences without requiring modification to the core code.

**Comment by jherico** [comment]
This should probably also eliminate the scripts in script-archive that use `Controller.createInputController`

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1077/HighFidelity-Beta-PR10797-384e2cefdb098e557fd801ce5d83b259399ec826.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1077/HighFidelity-Beta-PR10797-384e2cefdb098e557fd801ce5d83b259399ec826.exe

**Comment by ctrlaltdavid** [comment]
I think we need to delete the scripts in script-archive that use the now no longer available `Controller.createInputController` and friends.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1094/HighFidelity-Beta-PR10797-bed0e8ef8717a0c719e4c8fb5110e243fa946dc6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1094/HighFidelity-Beta-PR10797-bed0e8ef8717a0c719e4c8fb5110e243fa946dc6.exe

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1107/HighFidelity-Beta-PR10797-bed0e8ef8717a0c719e4c8fb5110e243fa946dc6.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1107/HighFidelity-Beta-PR10797-bed0e8ef8717a0c719e4c8fb5110e243fa946dc6.dmg

**Comment by Nex-Pro** [comment]
Testing this with leap motion.

**Comment by Nex-Pro** [comment]
# 10797 Make Leap Motion a plugin

Primary testing platform: Intel workstation.
OS: Win10 pro 64
CPU: i7 4.0 ghz
RAM: 16gb
GPU: Nvidia GTX1070 8gb VRam
HMD: Htc Vive.
Leap motion controller. 2.3.1 runtime.

Audio systems:  USB headset + Vive audio.

## QA Test PASS

Let me know if this needs a smoke test. 

**QA Tester**: Piper.Peppercorn

Leap Motion 2.3.1 runtime installed (i.e., not "Orion" beta). **CHECK**

Run Sandbox and Interface from PR build. **CHECK**

Use an avatar with individual features (e.g., "Being of Light").  **CHECK**

Settings > General > Leap Motion > Enabled **DONE**

Enables / disables Leap Motion control of hands. **SUCCESS**
Setting is remembered between Interface runs. **SUCCESS**

Settings > General > Leap Motion > Sensor location 

    Sets up Interface to use Leap Motion on desktop / HMD. **SUCCESS**
    Setting is remembered between Interface runs. **SUCCESS**

Settings > General > Leap Motion > Desktop height for horizontal forearms (default 0.20m) Seems inaccurate at higher setings. Fine at or below 0.2m.

    Sets the height of your hands above the Leap Motion that corresponds to your avatar's forearms being horizontal.  **SUCCESS**
    Setting is remembered between Interface runs. **SUCCESS**

Leap Motion controls position and orientation of hands.  **SUCCESS**

Leap Motion controls fingers and thumbs. **SUCCESS**

When tracking of a hand is lost,
your avatar's arm, hand, and fingers return to their relaxed,
non-controlled positions: hand and fingers resume with voice-activated gesturing (i.e., movements as you speak).  **SUCCESS**

Check another person (running PR Interface) can see your avatars' hands and fingers moving. **SUCCESS**

Test changing between desktop and HMD operation while Interface is running. **SUCCESS**

Test connecting Leap Motion after Interface has started. **SUCCESS**

Test enabling Leap Motion support in Settings and connecting Leap Motion after Interface has started. **SUCCESS**

**Comment by ZappoMan** [comment]
@ctrlaltdavid @jherico - why are we getting rid of InputController???

**Comment by ctrlaltdavid** [comment]
@ZappoMan Because there are no scripts that use it. Except some old ones in script-archive.

**Pull request merged**